modules/ROOT/pages/field-level-capabilities.adoc

```diff
@@ -0,0 +1,256 @@
+value providers for parameter fields, or to use parameter fields as acting
+parameters for either a value provider or sample data.
+
+These features are available for Mule 4.4.0, however, connectors can use
+the features without upgrading their minimum Mule version.
+
+== Defining a Value Provider for a Part of a Parameter
+
+You can declare a value provider for a parameter field by using the
+annotation *@FieldValues*.
+
+The *@FieldValues* annotation requires:
+
+* *value*
++
+Provides values for a field.
+* *targetSelectors*
++
+Describes which part of a parameter contains the values that are given by the value provider.
+It is defined through the DataWeave syntax to describe the selector through
+nested fields, XML attributes, or fields that contain spaces in their names.
+
+The following is an example of an operation that includes values provided for
+a parameter named `body`, which is a JSON input stream:
+
+----
+public class ChatOperations {
+ public void sendChatMessage(@Connection ChatConnection chatConnection,
+       @FieldValues(value = WorkspacesValueProvider.class, targetSelectors = "routingInfo.workspace")
+       @TypeResolver(ChatMessageResolver.class) @Content InputStream body) {
+    // Do message sending
+  }
+}
+----
+
+The `body` parameter has a field named *routingInfo*, which has a field named
+`workspace`. *WorkspacesValueProvider* is a value provider that uses the
+connection and retrieves possible workspaces.
+
+== Defining a Value Provider for Several Parts of a Parameter
+
+Parameters can have more than one part where values are provided.
+The *@FieldValues* annotation is a repeatable annotation, so parameters can be
+annotated multiple times.
+
+The following is an example of an operation with a parameter named
+`body`, which is a JSON input stream that has two parts with values that
+can be provided:
+
+----
+public class ChatOperations {
+ public void sendChatMessage(@Connection ChatConnection chatConnection,
+       @FieldValues(value = WorkspacesValueProvider.class, targetSelectors = "routingInfo.workspace")
+       @FieldValues(value = ChannelIdValueProvider.class, targetSelectors = "channelId")
+       @TypeResolver(ChatMessageResolver.class) @Content InputStream body) {
+    // Do message sending
+  }
+}
+----
+
+The parameter `body` defines values for two fields (*routingInfo.workspace* and
+*channelId*) that have values that are independent of each other.
+
+== Defining a Multilevel Value Provider for Several Parts of a Parameter
+
+You can use multilevel value providers for several parts of a parameter.
+
+Suppose in the previous example you want the values for the *channelId* field
+to depend on the values for the *routingInfo.workspace* field.
+
+To define this, the *targetSelectors* attribute from the *@FieldValues* annotation
+must receive an array of strings. If more than one string is given, a multilevel
+value provider is defined. Each given field defines the value parts of the
```

Comment:
Reword to active rather than passive voice, if possible, following the examples in previous comments.